packages/core/src/application.ts

```diff
@@ -88,6 +91,9 @@ export class Application extends Context {
     response: ServerResponse,
   ) {
     this._setupHandlerIfNeeded();
+    if (request.method === 'GET' && request.url === '/openapi.json') {
+      return this._serveOpenApiSpec(request, response);
```

Comment:
> I'm surprised you can't use the same methods from a sequence to do the route matching.

What methods do you mean? Are you referring to `findRoute`? The action `findRoute` is searching a table of all registered routes to find a handler (or a Controller method). To use that, I would have to register a new `Route`, which comes with the side effects I described in my comment above.

Also keep in mind that route matching is executed as part of the sequence. Here is the code handling http requests ([link](https://github.com/strongloop/loopback-next/blob/7888c6d0b7e6fd60f2e0a03fe28e83a7fad96ec8/packages/core/src/http-handler.ts#L54-L68)):

```ts
  protected async _handleRequest(
    request: ServerRequest,
    response: ServerResponse,
  ): Promise<void> {
    const parsedRequest: ParsedRequest = parseRequestUrl(request);
    const requestContext = this._createRequestContext(request, response);

    this._bindFindRoute(requestContext);
    this._bindInvokeMethod(requestContext);
    this._bindGetFromContext(requestContext);
    this._bindBindElement(requestContext);

    const sequence: SequenceHandler = await requestContext.get('sequence');
    await sequence.handle(parsedRequest, response);
}
```

> This just seems fragile. 

I agree this is not great. I think it's the cost of skipping Express and adding our Sequence-based handlers directly on top of core HTTP server.

> We'll certainly want to allow the url to be configurable and to support yaml.

Sure, I think we should allow an option to disable this endpoint too. In the code proposed here, the url can be configured via application config.

I am not sure why would you like an HTTP endpoint to return YAML representation of OpenAPI spec. A JSON document is a valid YAML document too, therefore the current JSON response can be loaded by any YAML-capable OpenAPI tool.

I think YAML is good for storing the spec into a file for further edit and use in e.g. API Connect. To support that use case, I prefer to create a CLI tool. This tool can call `app.getApiSpec()` and then save the object to a YAML file.

---

Let's focus on what's needed for the first Beta release. Can we land the version proposed by this pull request? If not, then what changes are you asking for?